homeassistant/components/media_player/songpal.py

```diff
@@ -0,0 +1,242 @@
+        return self._available
+
+    @property
+    def dev(self):
+        """Property for accessing the device handle."""
+        return self._dev
+
+    async def async_set_sound_setting(self, name, value):
+        """Change a setting on the device."""
+        res = await self.dev.set_sound_settings(name, value)
+
+        return res
+
+    async def async_update(self):
+        """Fetch updates from the device."""
+        from songpal import SongpalException
+        if not self._initialized:
+            try:
+                await self._dev.get_supported_methods()
+                self._sysinfo = await self.dev.get_system_info()
+                self._initialized = True
+            except SongpalException as ex:
+                _LOGGER.error("Unable to get methods from songpal: %s", ex)
+                raise PlatformNotReady
+
+        try:
+            volumes = await self.dev.get_volume_information()
+            if not volumes:
+                _LOGGER.error("Got no volume controls, bailing out")
+                return
+            if len(volumes) > 1:
+                _LOGGER.warning("Got %s volume controls, using the first one",
+                                volumes)
+                return
+
+            volume = volumes.pop()
+            _LOGGER.debug("Current volume: %s", volume)
+            self._volume_max = volume.maxVolume
+            self._volume_min = volume.minVolume
+            self._volume = volume.volume
+            self._volume_control = volume
+            self._is_muted = self._volume_control.is_muted
+
+            status = await self.dev.get_power()
+            self._state = status.status
+            _LOGGER.debug("Got state: %s", status)
+
+            inputs = await self.dev.get_inputs()
+            _LOGGER.debug("Got ins: %s", inputs)
+            self._sources = inputs
+
+            self._available = True
+        except SongpalException as ex:
+            # if we were available, print out the exception
+            if self._available:
+                _LOGGER.error("Got an exception %s", ex, exc_info=True)
+            self._available = False
+
+    async def async_select_source(self, source):
+        """Select source."""
+        for out in self._sources:
+            if out.title == source:
+                await out.activate()
+                return
+
+        _LOGGER.error("Unable to find output: %s", source)
+
+    @property
+    def source_list(self):
+        """Return list of available sources."""
+        return [x.title for x in self._sources]
+
+    @property
+    def state(self):
+        """Return current state."""
+        if self._state:
+            return STATE_ON
+        return STATE_OFF
+
+    @property
+    def source(self):
+        """Return currently active source."""
+        for out in self._sources:
+            if out.active:
+                return out.title
+
+        _LOGGER.error("Unable to find active output!")
```

Comment:
properties shouldn't log.

Also, inconsistent return, add `Return None`

Reply:
Removed the loggers (also the one below) from properties and added return here.